lib/src/element_type.dart

```diff
@@ -21,42 +21,38 @@ class ElementType {
 
   bool get isFunctionType => (_type is FunctionType);
 
-  bool get isParameterizedType {
-    if (_type is FunctionType) {
-      return typeArguments.isNotEmpty;
-    } else if (_type is ParameterizedType) {
-      return (_type as ParameterizedType).typeArguments.isNotEmpty;
-    }
-    return false;
-  }
+  bool get isParameterizedType => (_type is ParameterizedType);
 
   bool get isParameterType => (_type is TypeParameterType);
 
   String get linkedName {
-    if (_linkedName != null) return _linkedName;
-
-    StringBuffer buf = new StringBuffer();
-
-    if (isParameterType) {
-      buf.write(name);
-    } else {
-      buf.write(element.linkedName);
-    }
+    if (_linkedName == null) {
+      StringBuffer buf = new StringBuffer();
 
-    // not TypeParameterType or Void or Union type
-    if (isParameterizedType) {
-      if (typeArguments.every((t) => t.linkedName == 'dynamic')) {
-        _linkedName = buf.toString();
-        return _linkedName;
+      if (isParameterType) {
```

Comment:
`buf.write(isParameterType ? name : element.linkedName)`?

Reply:
This is all code from before that was indented.  Leaving.